### PR TITLE
interp_qed_delayed_proof doesn't need library internalization

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: vernac/synterp.ml [hunk 1]
```diff
@@ -409,9 +409,19 @@ let with_succeed f =
   Vernacstate.Synterp.unfreeze st;
   ControlSucceed { st = transient_st } :: ctrl, v
 
+let synpure_control : control_flag -> control_entry =
```
Comment (ejgallego) [inline vernac/synterp.ml:412]: Wouldn't it be better to share the code with `synterp_control_flag` ?

Comment (SkySkimmer) [reply vernac/synterp.ml:412]: not that easy, synterp_control_flag needs to handle the case with Fail where the command fails in synterp phase so the interp phase is turned into a noop

Not worth the effort imo

Comment (ejgallego) [reply vernac/synterp.ml:412]: I agree with you in terms of effort, let's stay like this for now indeed.

But the whole API setup is not good, we should really classify the commands with types.

**Review by ejgallego — APPROVED** [review]
LGTM thanks!

**Review by ejgallego — APPROVED** [review]
LGTM thanks!

**Pull request merged**